torchtune/datasets/multimodal/_llava_instruct.py

```diff
@@ -0,0 +1,113 @@
+    column_map: Optional[Dict[str, str]] = None,
+    new_system_prompt: Optional[str] = None,
+    image_tag: Optional[str] = "<image>",
+    train_on_input: bool = True,
+    packed: bool = False,
+    split: str = "train",
+    data_files: str = "llava_instruct_150k.json",
+    **load_dataset_kwargs: Dict[str, Any],
+) -> Union[SFTDataset, PackedDataset]:
+    """
+    Support for family of image + text datasets similar to
+    `LLaVA-Instruct-150K <https://huggingface.co/datasets/liuhaotian/LLaVA-Instruct-150K>_`
+    from Hugging Face Datasets.
+
+    To use this dataset, you must first download the COCO Train 2017 image dataset. You can do so
+    by visiting https://cocodataset.org/#download or downloading it directly:
+
+    .. code-block:: bash
+
+        wget -c http://images.cocodataset.org/zips/train2017.zip
+        unzip train2017.zip -d coco/
+
+    Then, you must pass in the directory containing all the images to the ``image_dir`` argument
+    so each file can be loaded as PIL images in the transform. In the example above, you would
+    set ``image_dir="coco/train2017"``.
+
+    Args:
+        model_transform (Transform): model-specific transform that takes in a sample dict and applies custom
+            transforms on the keys. The tokenizer used by the model should be encapsulated in the model transform
+            and should operate on the "messages" field. Any model-specific image transforms should operate on
+            the "images" field. The keys returned by the model should be aligned with the
+            expected inputs into the model.
+        coco_image_dir (str): path to the directory containing the COCO Train 2017 images.
+        source (str): path to dataset repository on Hugging Face. For local datasets,
+            define source as the data file type (e.g. "json", "csv", "text") and pass
+            in the filepath in ``data_files``. See `Hugging Face's
+            <https://huggingface.co/docs/datasets/en/package_reference/loading_methods#datasets.load_dataset.path>`_
+        column_map (Optional[Dict[str, str]]): a mapping from the expected columns ("conversations")
+            to the new column names in the dataset. If None, assume these are identical.
+            Default is None.
+        new_system_prompt (Optional[str]): if specified, prepend a system message. This can
+            serve as instructions to guide the model response. Setting this will OVERRIDE any system
+            messages already present in the dataset. Default is None.
+        image_tag (Optional[str]): if specified, split the raw text content by the specified ``image_tag``
+            and use placeholders for where the images are present in the text for proper tokenization. Set
+            this if your dataset contains images and uses a specific string (ex: "<image>") to indicate the
+            presence of an image. Leave this as None if your dataset does not contain images. Default is "<image>".
```

Comment:
When would one of these not contain images though?

Reply:
mmm... I guess you're right, I suppose it doesn't make sense to make this configurable